ingestionpolicies.go

```diff
@@ -4,15 +4,46 @@ import (
 	"fmt"
 )
 
-const basePolicyPath = "/api/v2/usage/ingestionpolicy"
-
-type IngestionPolicy struct {
-	ID                  string `json:"id,omitempty"`
-	Name                string `json:"name"`
-	Description         string `json:"description"`
-	UserAccountCount    int    `json:"userAccountCount"`
-	ServiceAccountCount int    `json:"serviceAccountCount"`
-	Scope               string `json:"scope"`
+const baseIngestionPolicyPath = "/api/v2/usage/ingestionpolicy"
+
+type IngestionPolicyAccount struct {
+	Name string `json:"name,omitempty"`
+	ID   string `json:"id,omitempty"`
+}
+
+type IngestionPolicyGroup struct {
+	Name        string `json:"name,omitempty"`
+	ID          string `json:"id,omitempty"`
+	Description string `json:"description,omitempty"`
+}
+
+type IngestionPolicyTag struct {
+	Key   string `json:"key,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type IngestionPolicyResponse struct {
+	ID          string                   `json:"id,omitempty"`
+	Name        string                   `json:"name,omitempty"`
+	Description string                   `json:"description,omitempty"`
+	Scope       string                   `json:"scope,omitempty"`
+	Accounts    []IngestionPolicyAccount `json:"accounts,omitempty"`
+	Groups      []IngestionPolicyGroup   `json:"groups,omitempty"`
+	Sources     []string                 `json:"sources,omitempty"`
+	Namespaces  []string                 `json:"namespaces,omitempty"`
+	Tags        []IngestionPolicyTag     `json:"pointTags,omitempty"`
+}
+
+type IngestionPolicyRequest struct {
+	ID          string              `json:"id,omitempty"`
+	Name        string              `json:"name,omitempty"`
+	Description string              `json:"description,omitempty"`
+	Scope       string              `json:"scope,omitempty"`
+	Accounts    []string            `json:"accounts,omitempty"`
+	Groups      []string            `json:"groups,omitempty"`
+	Sources     []string            `json:"sources,omitempty"`
+	Namespaces  []string            `json:"namespaces,omitempty"`
+	Tags        []map[string]string `json:"pointTags,omitempty"`
```

Comment:
I honestly don't know, that's just what the API (Swagger) docs say. 😅
```
"pointTags": [
    {
      "key": "string",
      "value": "string"
    }
  ]
```
It does seem weird... I could ask the API team.

Thinking about it a bit. I wonder if it's because they want to support a broader set of characters as keys. If I remember correctly, JSON values support more characters than the keys (not confident about that tho). 🤔

Reply:
I guess if swagger says the API should look like this then you are right. I just can't think of any use cases when you would want more than one map. Unless as you say maybe a key could map to multiple values but if that is true why not just use map[string][]string? It might be worth asking the API team.